storage/badger/blocks.go

```diff
@@ -52,7 +52,25 @@ func (b *Blocks) retrieveTx(blockID flow.Identifier) func(*badger.Txn) (*flow.Bl
 		if err != nil {
 			return nil, fmt.Errorf("could not retrieve payload: %w", err)
 		}
-		return flow.NewBlock(header.HeaderBody, *payload), nil
+
+		untrustedBlock := flow.UntrustedBlock{
```

Comment:
This is a bit awkward, since the `b.payloads` and `b.headers` will have already validated both fields of the block. But I'm not sure how to make it any better while keeping the same constructor safety properties 😅. 

Our in-place validation checks are inexpensive, so my assumption is that any extra cycles spent will be negligible.